### PR TITLE
treewide: fix or enable darwin build for many packages

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
This is going to be one of many PRs. I created a [query script](https://github.com/siraben/nixpkgs/commit/b3c5aa660d43d2c22d38bf8d14c0e3bca8495ea6) to be able to find packages with no buildInputs that don't have darwin in the `meta.platforms`. Sometimes this was due to a lack of having a `meta.platforms` at all, but in many cases simply expanding the `meta.platforms` to include darwin fixed the issue. Otherwise I had to do minor changes to get it to build (hence why I chose packages with `buildInputs = []`).

Over 1000 packages were returned by my script and so it would be better to do this in small PRs at a time.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/serpent/default.nix [hunk 1]
```diff
@@ -33,6 +37,6 @@ stdenv.mkDerivation {
     homepage = "https://github.com/ethereum/wiki/wiki/Serpent";
     license = with licenses; [ wtfpl ];
     maintainers = with maintainers; [ chris-martin ];
-    platforms = with platforms; linux;
+    platforms = with platforms; all;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/serpent/default.nix:40]:
```suggestion
    platforms = platforms.all;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114976` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>_4th</li>
<li>mpage</li>
<li>muon</li>
<li>rasm</li>
<li>serpent</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
_4th:
<details>


warning: unclear-gpl
`lgpl3` is a deprecated license, check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/development/compilers/4th/default.nix:36:5:

```
   |
36 |     license = licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/compilers/4th/default.nix:33:3:

```
   |
33 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
error: no-flags-spaces
`makeFlags` cannot contain spaces, please use `makeFlagsArray` in Bash.

Near pkgs/development/compilers/4th/default.nix:14:3:

```
   |
14 |   makeFlags = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-spaces.md
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114976` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
14 packages built:
</summary>
<ul>
<li>QuadProgpp</li>
<li>_0x0</li>
<li>_4th</li>
<li>arachne-pnr</li>
<li>gprolog</li>
<li>miranda</li>
<li>mpage</li>
<li>muon</li>
<li>numdiff</li>
<li>rasm</li>
<li>serpent</li>
<li>snooze</li>
<li>uni2ascii</li>
<li>yaml-merge</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
QuadProgpp:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/science/math/QuadProgpp/default.nix:23:5:

```
   |
23 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
_4th:
<details>

error: no-flags-spaces
`makeFlags` cannot contain spaces, please use `makeFlagsArray` in Bash.

Near pkgs/development/compilers/4th/default.nix:14:3:

```
   |
14 |   makeFlags = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-spaces.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/compilers/4th/default.nix:33:3:

```
   |
33 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
warning: unclear-gpl
`lgpl3` is a deprecated license, check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/development/compilers/4th/default.nix:36:5:

```
   |
36 |     license = licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
arachne-pnr:
<details>


warning: patch-phase
`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/compilers/arachne-pnr/default.nix:22:3:

```
   |
22 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
</details>
gprolog:
<details>


warning: patch-phase
`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/compilers/gprolog/default.nix:16:3:

```
   |
16 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
</details>
mpage:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/text/mpage/default.nix:15:3:

```
   |
15 |   meta = {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
numdiff:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/text/numdiff/default.nix:23:5:

```
   |
23 |     maintainers = with maintainers; [];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
rasm:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/compilers/rasm/default.nix:30:5:

```
   |
30 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
uni2ascii:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/tools/text/uni2ascii/default.nix:11:5:

```
   |
11 |     license = lib.licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Comment by siraben** [comment]
Fixup'd your suggestion into the serpent commit.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114976` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
12 packages built:
</summary>
<ul>
<li>QuadProgpp</li>
<li>_4th</li>
<li>arachne-pnr</li>
<li>gprolog</li>
<li>mpage</li>
<li>muon</li>
<li>numdiff</li>
<li>rasm</li>
<li>serpent</li>
<li>snooze</li>
<li>uni2ascii</li>
<li>yaml-merge</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
QuadProgpp:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/science/math/QuadProgpp/default.nix:23:5:

```
   |
23 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
_4th:
<details>

error: no-flags-spaces
`makeFlags` cannot contain spaces, please use `makeFlagsArray` in Bash.

Near pkgs/development/compilers/4th/default.nix:14:3:

```
   |
14 |   makeFlags = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-spaces.md
warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/compilers/4th/default.nix:33:3:

```
   |
33 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
warning: unclear-gpl
`lgpl3` is a deprecated license, check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/development/compilers/4th/default.nix:36:5:

```
   |
36 |     license = licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
arachne-pnr:
<details>


warning: patch-phase
`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/compilers/arachne-pnr/default.nix:22:3:

```
   |
22 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
</details>
gprolog:
<details>


warning: patch-phase
`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/development/compilers/gprolog/default.nix:16:3:

```
   |
16 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
</details>
mpage:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/text/mpage/default.nix:15:3:

```
   |
15 |   meta = {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
numdiff:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/text/numdiff/default.nix:23:5:

```
   |
23 |     maintainers = with maintainers; [];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
rasm:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/compilers/rasm/default.nix:30:5:

```
   |
30 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
uni2ascii:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/tools/text/uni2ascii/default.nix:11:5:

```
   |
11 |     license = lib.licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Pull request merged**